### PR TITLE
[DOC] Added warning about quoting in `.env` file for Docker commands

### LINKED ISSUE #224: Warn about quotes interpretation in README instructions for using API docker image directly
In the below sections of the README, we should explicitly instruct users to not wrap any space-separated values for variables in the .env file in quotation marks:
- [x] https://github.com/neurobagel/api#option-2-use-the-latest-image-from-docker-hub
- [x] https://github.com/neurobagel/api#option-3-build-the-image-using-the-dockerfile
(for now, it's probably just the `NB_API_ALLOWED_ORIGINS` variable that this applies to)

See our internal wiki https://github.com/neurobagel/documentation/wiki/Docker#docker-cli-docker-run for why the quotes are a problem

Originally posted by @surchs in https://github.com/neurobagel/federation-api/pull/22#pullrequestreview-1732149368

### PR DESCRIPTION
**Pull request by samadpls**
- Closes #224 

Changes proposed in this pull request:

- Added warning about quoting in `.env` file for Docker commands

## Checklist

- [x] PR has an interpretable title with a prefix (`[ENH]`, `[FIX]`, `[REF]`, `[TST]`, `[CI]`, `[MNT]`, `[INF]`, `[MODEL]`, `[DOC]`) _(see https://neurobagel.org/contributing/pull_requests for more info)_
- [x] PR has a label for the release changelog or `skip-release` (to be applied by maintainers only)
- [x] PR links to GitHub issue with mention `Closes #XXXX`
- [ ] Tests pass
- [x] Checks pass

For new features:
- [ ] Tests have been added

For bug fixes:
- [ ] There is at least one test that would fail under the original bug conditions.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -106,6 +106,7 @@ If you wish to set different port numbers, modify your `.env` file accordingly a
 export $(cat .env | xargs)  # export your .env file to expose your set port numbers to the -p flag of docker run
 docker run -d --name=api -p ${NB_API_PORT_HOST}:${NB_API_PORT} --env-file=.env neurobagel/api
 ```
+> :warning: **IMPORTANT:** Do not wrap any space-separated values for variables in the `.env` file in quotation marks, as it may lead to issues.
```
Comment (alyssadai) [inline README.md:109]:
```suggestion
> :warning: **IMPORTANT:** If using the above command, do not wrap any values for variables in the `.env` file in quotation marks, as they will be interpreted literally and may lead to [issues](https://github.com/docker/for-linux/issues/1208).
```

File: README.md [hunk 2]
```diff
@@ -120,6 +121,7 @@ docker build -t neurobagel/api .
...
```
Comment (alyssadai) [inline README.md:124]:
```suggestion
> :warning: **IMPORTANT:** If using the above command, do not wrap any values for variables in the `.env` file in quotation marks, as they will be interpreted literally and may lead to [issues](https://github.com/docker/for-linux/issues/1208).
```

**Review by alyssadai — APPROVED** [review]
Thanks a lot for addressing this @samadpls, and for the nice formatting of the warnings! 😄

A minor phrasing suggestion below for clarity (I think it's always helpful to link to existing issues where possible). Otherwise, this should be good to merge! 🧑‍🍳

**Review by alyssadai — APPROVED** [review]

**Comment by surchs** [comment]
:rocket: PR was released in [`v0.2.0`](https://github.com/neurobagel/api/releases/tag/v0.2.0) :rocket:

**Pull request merged**